tests/plugins/test_pulp_pull.py

```diff
@@ -528,11 +530,121 @@ def get_manifest(request):
         workflow.postbuild_plugins_conf = [{'name': 'pulp_sync'}]
         tasker = MockerTasker()
         plugin = PulpPullPlugin(tasker, workflow, timeout=0.2,
-                                retry_delay=0.1, expect_v2schema2=True)
+                                retry_delay=0.1, expect_v2schema2=expect_v2)
         media_types = plugin.run()
 
         expected_media_types = [MEDIA_TYPE_DOCKER_V2_MANIFEST_LIST]
         if not manifest_list_only:
             expected_media_types.append(MEDIA_TYPE_DOCKER_V2_SCHEMA1)
             expected_media_types.append(MEDIA_TYPE_DOCKER_V2_SCHEMA2)
+        else:
+            assert "Only V2 schema 2 manifest list is expected, " in caplog.text()
         assert set(media_types) == set(expected_media_types)
+
+    @responses.activate
+    @pytest.mark.parametrize(('found_v2', 'found_v2_list'), [
```

Comment:
How about:
```
@pytest.mark.parametrize(('platforms', 'has_v1', 'has_v2', 'has_v2_list', 'expect_v1', 'expect_v2', 'expect_v2_list')
```

This will make it easier to tell what the different scenarios being tested are.

Reply:
sure can do that, I just wanted to cover all v2  & v2 list cases
as v1 isn't very useful variable in this test

Reply:
also expect_v1 is not exactly we care about at all in pulp_pull